tests/e2e.test.ts

```diff
@@ -0,0 +1,262 @@
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import * as dotenv from 'dotenv';
+import { Contract, ethers } from 'ethers';
+import { JsonRpcSigner, Web3Provider } from '@ethersproject/providers';
+import { Server } from '@chainlink/ccip-read-server';
+import { dohQuery } from '@ensdomains/dnsprovejs';
+import * as packet from 'dns-packet';
+import supertest from 'supertest';
+import { makeApp } from '../src/app';
+import OffchainDNSResolver_abi from '@ensdomains/ens-contracts/artifacts/contracts/dnsregistrar/OffchainDNSResolver.sol/OffchainDNSResolver.json';
+import Resolver_abi from '@ensdomains/ens-contracts/artifacts/contracts/resolvers/OwnedResolver.sol/OwnedResolver.json';
+import {
+  BaseProvider,
+  BlockTag,
+  TransactionRequest,
+  Network,
+} from '@ethersproject/providers';
+import { fetchJson } from '@ethersproject/web';
+import { arrayify, BytesLike, hexlify } from '@ethersproject/bytes';
+dotenv.config();
+chai.use(chaiAsPromised);
+
+export type Fetch = (url: string, json?: string) => Promise<any>;
+
+const Resolver = new ethers.utils.Interface(Resolver_abi.abi);
+
+const ENS_ADDRESS = '0xE6E340D132b5f46d1e472DebcD681B2aBc16e57E';
+const DNSSEC_IMPL = '0xB7f8BC63BbcaD18155201308C8f3540b07f84F5e';
+const OWNED_RESOLVER = '0x5eb3Bc0a489C5A8288765d2336659EbCA68FCd00';
+
+const TEST_URL = 'https://localhost:8000/query';
+const TEST_NAME = 'tanrikulu.xyz'; // use a domain with resolver txt set (e.g. ENS1 0x5eb3Bc0a489C5A8288765d2336659EbCA68FCd00)
+const TEST_ADDRESS = '0xfefeFEFeFEFEFEFEFeFefefefefeFEfEfefefEfe';
+
+const CCIP_READ_INTERFACE = new ethers.utils.Interface(
+  OffchainDNSResolver_abi.abi
+);
+
+function deploySolidity(data: any, signer: ethers.Signer, ...args: any[]) {
+  const factory = ethers.ContractFactory.fromSolidity(data, signer);
+  return factory.deploy(...args);
+}
+
+export class MockProvider extends BaseProvider {
+  readonly parent: BaseProvider;
+  readonly fetcher: Fetch;
+
+  /**
+   * Constructor.
+   * @param provider: The Ethers provider to wrap.
+   */
+  constructor(provider: BaseProvider, fetcher: Fetch = fetchJson) {
+    super(31337);
+    this.parent = provider;
+    this.fetcher = fetcher;
+  }
+
+  async perform(method: string, params: any): Promise<any> {
+    switch (method) {
+      case 'call':
+        const { result } = await this.handleCall(this, params);
+        return result;
+      default:
+        return this.parent.perform(method, params);
+    }
+  }
+
+  async handleCall(
+    provider: MockProvider,
+    params: { transaction: TransactionRequest; blockTag?: BlockTag }
+  ): Promise<{ transaction: TransactionRequest; result: BytesLike }> {
+    let result = await provider.parent.perform('call', params);
+
+    if (!result.startsWith('0x556f1830')) {
```

Comment:
You also need to check that `result.length % 128 == 10` to guard against the (unlikely) case that a valid response begins with a function selector.